receiver/vmmetricsreceiver/vm_metrics_collector.go

```diff
@@ -51,6 +53,8 @@ const (
 	defaultScrapeInterval = 10 * time.Second
 )
 
+var rsc *resourcepb.Resource
```

Comment:
why not as a field of `VMMetricsCollector`? is it to be shared between all instances?

Reply:
It should be shared. It is associated with the agent rather than the VmMetricReceiver.